### PR TITLE
[Inductor UT] Generalize newly introduced inductor UTs for intel GPU (Part 2)

### PR DESCRIPTION
**Pull request by hoshibara**
[Inductor UT] Generalize Newly introduced inductor UTs for intel GPU
reuse `test/inductor/test_inductor_freezing.py`
reuse `test/inductor/test_layout_optim.py`
reuse `test/inductor/test_loop_ordering.py`
reuse `test/inductor/test_memory_planning.py`
reuse `test/inductor/test_padding.py`

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: test/inductor/test_memory_planning.py [hunk 1]
```diff
@@ -79,6 +92,7 @@ def test_cpp_wrapper(self):
         self.assertTrue(same(f(*args), result))
 
     @skipIfRocm(msg="test_aot_inductor doesn't work on ROCm")
+    @skipIfXpu
```
Comment (EikanWang) [inline test/inductor/test_memory_planning.py:95]: Message?

File: test/inductor/test_inductor_freezing.py [hunk 2]
```diff
@@ -956,14 +962,22 @@ class FreezingCpuTests(TestCase):
 
     copy_tests(OptimizeForInferenceTemplate, FreezingCpuTests, "cpu")
 
-if HAS_CUDA and not TEST_WITH_ASAN:
+if HAS_GPU and not TEST_WITH_ASAN:
+    if GPU_TYPE == "cuda":
+
+        class FreezingCudaTests(TestCase):
```
Comment (etaf) [inline test/inductor/test_inductor_freezing.py:968]: FreezingCudaTests =. FreezingGpuTests

File: test/inductor/test_inductor_freezing.py [hunk 3]
```diff
@@ -956,14 +962,22 @@ class FreezingCpuTests(TestCase):
...
+            common = check_model_cuda
```
Comment (etaf) [inline test/inductor/test_inductor_freezing.py:969]: check_model_cuda => check_model_gpu

File: test/inductor/test_loop_ordering.py [hunk 4]
```diff
@@ -234,6 +237,7 @@ def f(x, y):
         expected_num_bytes *= x.itemsize
         self.assertEqual(expected_num_bytes, metrics.num_bytes_accessed)
 
+    @expectedFailureXPU
```
Comment (etaf) [inline test/inductor/test_loop_ordering.py:240]: Any comment  for the expecedFailure?

File: test/inductor/test_torchinductor_opinfo.py [hunk 5]
```diff
@@ -356,6 +356,8 @@ def format_op(op):
     "nn.functional.multilabel_margin_loss": {f16},
     "nn.functional.multi_margin_loss": {f16},
     "nn.functional.avg_pool3d": {f16},
+    # not implemented for 'Bool'
+    "nn.functional.unfold": {b8},
```
Comment (EikanWang) [inline test/inductor/test_torchinductor_opinfo.py:360]: https://github.com/pytorch/pytorch/pull/137873 will cover this change.

**Review by EikanWang — APPROVED** [review]

File: torch/_inductor/ir.py [hunk 6]
```diff
@@ -3826,7 +3826,8 @@ def simplify_and_reorder(x_vars, support_vars, sizes, simplify_loops):
 
         support_vars = index_vars + reduce_vars
         should_merge_loops = (
-            self.get_device().type != "cuda" or not config.loop_ordering_after_fusion
+            self.get_device().type not in ("cuda", "xpu")
```
Comment (etaf) [inline torch/_inductor/ir.py:3829]: not is_gpu(self.get_device().type)

**Review by etaf — CHANGES_REQUESTED** [review]

File: test/inductor/test_torchinductor_opinfo.py (hunk 5 above)
Comment (hoshibara) [reply test/inductor/test_torchinductor_opinfo.py:360]: Just for CI, I will resolve this conflict after #137873 is merged.

File: torch/_inductor/ir.py (hunk 6 above)
Comment (hoshibara) [reply torch/_inductor/ir.py]: Fixed

File: test/inductor/test_loop_ordering.py (hunk 4 above)
Comment (hoshibara) [reply test/inductor/test_loop_ordering.py:241]: Fixed

File: test/inductor/test_inductor_freezing.py (hunk 3 above)
Comment (hoshibara) [reply test/inductor/test_inductor_freezing.py]: Fixed

File: test/inductor/test_inductor_freezing.py (hunk 2 above)
Comment (hoshibara) [reply test/inductor/test_inductor_freezing.py]: Fixed

File: test/inductor/test_memory_planning.py (hunk 1 above)
Comment (hoshibara) [reply test/inductor/test_memory_planning.py]: Fixed

**Comment by etaf** [comment]
Please rebase to resolve the mrege conflicts.

**Comment by hoshibara** [comment]
> Please rebase to resolve the mrege conflicts.

Resolved

**Review by etaf — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**